lure.go

```diff
@@ -125,9 +135,23 @@ func runMain(config *project.LureConfig, auth vcs.Authentication) {
 }
 
 func mainWithEnvironmentAuth(config *project.LureConfig) {
-	auth := vcs.UserPassAuth{
-		Username: os.Getenv("BITBUCKET_USERNAME"),
-		Password: os.Getenv("BITBUCKET_PASSWORD"),
+	var auth vcs.Authentication
+	accessToken := os.Getenv("GITHUB_ACCESS_TOKEN")
+	if accessToken != "" {
+		auth = vcs.TokenAuth{User: "x-access-token", Token: accessToken}
+	} else {
+		username := os.Getenv("GITHUB_USERNAME")
```

Comment:
can we move this to the provider? GetUserName, GetPassword

Reply:
The problem is the auth is determined before the provider. So, it's possible, but that would necessitate a refactor.